### PR TITLE
Enhancement: API Update

### PR DESCRIPTION
**Pull request by sam1357**
- Updated MOCBOT to call the MOCBOT API for all database calls - including an API handler util
- Ported over the AFK module from MOCBOT v2 to MOCBOT

### CODE REVIEW

File: lib/cogs/Levels.py [hunk 1]
```diff
@@ -123,32 +141,28 @@ async def on_message(self, message):
...
+        xp_data = data or await self.get_xp_data(member)
+        member_level = xp_data.get("Level", None) if xp_data is not None else 0
         if member:
-            if (role_map := MOC_DB.field("SELECT LevelRoles FROM Roles WHERE GuildID = %s", member.guild.id)) != None:
-                role_map = self.keystoint(json.loads(role_map))
-                member_level = await self.get_level(member) or 0
+            res = API.get(f"/roles/{member.guild.id}")
+            role_map = res.get("LevelRoles", None) if res is not None else None
+            if role_map != None:
+                role_map = self.dict_to_int(role_map)
```
Comment (MasterOfCubesAU) [inline lib/cogs/Levels.py:165]: Perhaps merge the body of `dict_to_int()` here as it's a niche case?

File: lib/cogs/Lobbies.py [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
 class LobbyPrompt(View):
     def __init__(self, *, timeout=180, interaction: discord.Interaction):
         super().__init__(timeout=timeout)
-        self.lobby_category = interaction.client.get_channel(MOC_DB.field("SELECT LOBBY_CATEGORY FROM Guild_Settings WHERE GuildID = %s", interaction.guild.id))
+        settings = API.get(f'/settings/{interaction.guild.id}')
+        self.lobby_category = interaction.guild.get_channel(int(settings.get("LobbyCategory") if settings is not None else None))
         self.interaction = interaction
         self.updateOptions()
 
+    # def on_error(self, interaction: discord.Interaction, error: Exception, item: discord.ui.Item):
```
Comment (MasterOfCubesAU) [inline lib/cogs/Lobbies.py:20]: Remove

File: lib/cogs/Lobbies.py [hunk 3]
```diff
@@ -219,13 +243,26 @@ def check(message):
...
+                    try:    
+                        await self.invite_user(member, lobby_details, lobby_users)
+                    except (discord.errors.HTTPException, AttributeError):
+                        pass
+                    else:
+                        members_to_add.append(str(member.id))
+            if len(members_to_add) != 0:
+                API.post(f'/lobby/{self.interaction.guild.id}/{lobby_details.get("LeaderID")}/users', members_to_add)
             await self.updateView()
             # await interaction.followup.send(embed=self.interaction.client.create_embed("MOCBOT LOBBIES", f"Invited {' '.join([x.mention for x in msg.mentions])}", None), ephemeral=True)
```
Comment (MasterOfCubesAU) [inline lib/cogs/Lobbies.py:260]: Remvoe

File: lib/cogs/Lobbies.py [hunk 4]
```diff
@@ -239,14 +276,22 @@ def check(message):
...
-                await self.remove_user(member, lobby_details[1])
-                await member.send(embed=self.interaction.client.create_embed("MOCBOT LOBBIES", f"You have been kicked from **{lobby_details[1]}**", None))
+                if (not LobbyPrompt.is_lobby_user(member, lobby_details, lobby_users)) or LobbyPrompt.is_lobby_leader(member, lobby_details):
+                    pass
+                try:
+                    await self.remove_user(member, lobby_details)
+                except (discord.errors.HTTPException, AttributeError):
+                    pass 
             await self.updateView()
             # await interaction.followup.send(embed=self.interaction.client.create_embed("MOCBOT LOBBIES", f"Kicked {' '.join([x.mention for x in msg.mentions])}", None), ephemeral=True)
```
Comment (MasterOfCubesAU) [inline lib/cogs/Lobbies.py:289]: Remove

File: lib/cogs/Lobbies.py [hunk 5]
```diff
@@ -301,6 +356,8 @@ def __init__(self, LobbyPrompt) -> None:
     async def on_submit(self, interaction: discord.Interaction):
         # await self.LobbyPrompt.delete_prompt()
```
Comment (MasterOfCubesAU) [inline lib/cogs/Lobbies.py:357]: Remove

**Review by MasterOfCubesAU — CHANGES_REQUESTED** [review]

File: lib/cogs/Guilds.py [hunk 6]
```diff
@@ -26,7 +23,7 @@ async def on_guild_join(self, guild):
 
     @commands.Cog.listener()
     async def on_guild_remove(self, guild):
-        MOC_DB.execute("DELETE FROM Guild_Settings WHERE GuildID = %s", guild.id)
+        API.delete(f'/settings/{guild.id}')
```
[the remark this answers is not in the document]
Comment (sam1357) [reply lib/cogs/Guilds.py:26]: will fix later

**Review by MasterOfCubesAU — APPROVED** [review]

**Pull request merged**